### PR TITLE
add support for consul-k8s health checks controller for service mesh

### PR DESCRIPTION
**Pull request by kschoche**
Add support for enabling/disabling consul-k8s health check controller for service mesh which is responsible for updating Consul with pod readiness status changes.

Two new options are added : 
1 `connectInject.healthChecks.enabled` which enables/disables the controller at runtime
2 `connectInject.healthChecks.reconcilePeriod` which modifies the reconcile frequency for the controller.

Acceptance tests to validate end-to-end functionality are added as well as unit tests.

### CODE REVIEW

File: test/unit/connect-inject-clusterrole.bats [hunk 1]
```diff
@@ -108,6 +108,29 @@ load _helpers
...
 }
+
+@test "connectInject/ClusterRole: pod resource permission set when health checks are enabled" {
+  cd `chart_dir`
+  local actual=$(helm template \
+      -s templates/connect-inject-clusterrole.yaml  \
+      --set 'connectInject.enabled=true' \
+      --set 'connectInject.healthChecks.enabled=true' \
+      . | tee /dev/stderr |
+      yq -r '.rules[1].resources[0]' | tee /dev/stderr)
```
Comment (lkysow) [inline test/unit/connect-inject-clusterrole.bats:122]: can you use `map(select(.resources[0] == pods))` to match the changes above

File: test/unit/connect-inject-clusterrole.bats [hunk 2]
```diff
@@ -108,6 +108,29 @@ load _helpers
...
+}
+
+@test "connectInject/ClusterRole: no pod resource permission set when health checks are disabled" {
+  cd `chart_dir`
+  local actual=$(helm template \
+      -s templates/connect-inject-clusterrole.yaml  \
+      --set 'connectInject.enabled=true' \
+      --set 'connectInject.healthChecks.enabled=false' \
+      . | tee /dev/stderr |
+      yq -r '.rules | length' | tee /dev/stderr)
```
Comment (lkysow) [inline test/unit/connect-inject-clusterrole.bats:133]: Can you use the map select here and assert len == 0

File: test/unit/connect-inject-deployment.bats [hunk 3]
```diff
@@ -79,6 +79,61 @@ load _helpers
...
+@test "connectInject/Deployment: health checks reconcile period set by default" {
+  cd `chart_dir`
+  local cmd=$(helm template \
+      -s templates/connect-inject-deployment.yaml \
+      --set 'connectInject.enabled=true' \
+      . | tee /dev/stderr |
+      yq '.spec.template.spec.containers[0].command' | tee /dev/stderr)
+
+  local actual=$(echo "$cmd" |
+    yq 'any(contains("-health-checks-reconcile-period"))' | tee /dev/stderr)
```
Comment (lkysow) [inline test/unit/connect-inject-deployment.bats:118]:
```suggestion
    yq 'any(contains("-health-checks-reconcile-period=1m"))' | tee /dev/stderr)
```

File: test/unit/connect-inject-deployment.bats [hunk 4]
```diff
@@ -79,6 +79,61 @@ load _helpers
...
+      --set 'connectInject.healthChecks.enabled=true' \
+      . | tee /dev/stderr |
+      yq '.spec.template.spec.containers[0].command' | tee /dev/stderr)
+
+  local actual=$(echo "$cmd" |
+    yq 'any(contains("-enable-health-checks-controller=true"))' | tee /dev/stderr)
+  [ "${actual}" = "true" ]
+}
+
+@test "connectInject/Deployment: health checks reconcile period set by default" {
```
Comment (lkysow) [inline test/unit/connect-inject-deployment.bats:109]: Need test that reconcile period can be set.

**Review by lkysow — COMMENTED** [review]
Looks great!

**Review by lkysow — APPROVED** [review]
🎉

File: test/acceptance/tests/fixtures/cases/static-server-hc/patch.yaml [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+  name: static-server
+spec:
+  template:
+    metadata:
+      annotations:
+        "consul.hashicorp.com/connect-inject": "true"
+    spec:
+      containers:
+        - name: static-server
+          image: kschoche/http-echo:latest
```
Comment (ashwin-venkatesh) [inline test/acceptance/tests/fixtures/cases/static-server-hc/patch.yaml:13]: Did we talk about maybe swapping out this image?

**Review by thisisnotashwin — APPROVED** [review]
💥

File: test/acceptance/tests/fixtures/cases/static-server-hc/patch.yaml (hunk 5 above)
Comment (kschoche) [reply test/acceptance/tests/fixtures/cases/static-server-hc/patch.yaml:13]: had a chat with @lkysow on this topic and we're okay leaving it for now.
If we want to change it for GA I think that is also fine.

Comment (ashwin-venkatesh) [reply test/acceptance/tests/fixtures/cases/static-server-hc/patch.yaml:13]: ship it 😄

**Pull request merged**